src/js/components/MultiSelect/MultiSelect.js

```diff
@@ -0,0 +1,680 @@
+        if (
+          typeof nextValue !== 'object' &&
+          nextValue !== event.target.value &&
+          inputRef.current
+        ) {
+          // select registers changing option as a click event or keydown.
+          // when in a form, we need to programatically trigger a change
+          // event in order for the change event to be registered upstream
+          // necessary for change validation in form
+          triggerChangeEvent(nextValue);
+        }
+        setValue(nextValue);
+        if (onChange) {
+          event.persist();
+          let adjustedEvent;
+          // support for native event used by Preact
+          if (event instanceof Event) {
+            adjustedEvent = new event.constructor(event.type, event);
+            Object.defineProperties(adjustedEvent, {
+              target: { value: inputRef.current },
+              value: { value: nextValue },
+              option: { value: option },
+            });
+          } else {
+            adjustedEvent = event;
+            adjustedEvent.target = inputRef.current;
+            adjustedEvent.value = nextValue;
+            adjustedEvent.option = option;
+          }
+          onChange(adjustedEvent);
+        }
+      },
+      [onChange, setValue, triggerChangeEvent],
+    );
+
+    const SelectIcon = getSelectIcon(icon, theme, open);
+
+    const optionValue = useCallback(
+      (index) => getOptionValue(index, allOptions, valueKey),
+      [allOptions, valueKey],
+    );
+
+    const isDisabled = useCallback(
+      (index) =>
+        checkDisabled(index, disabled, disabledKey, allOptions, valueKey),
+      [disabled, disabledKey, allOptions, valueKey],
+    );
+
+    const visibleValue = useCallback(
+      (i) => {
+        const iValue = valueKey && valueKey.reduce ? applyKey(i, valueKey) : i;
+        const indexOptions = allOptions.indexOf(i);
+        const iLabel = getOptionLabel(indexOptions, allOptions, labelKey);
+        if (value.indexOf(iValue) < 5) {
+          let child;
+          if (children) {
+            const optionDisabled = isDisabled(indexOptions);
+            const optionSelected = true;
+            const optionActive = false;
+            child = children(i, indexOptions, allOptions, {
+              active: optionActive,
+              disabled: optionDisabled,
+              selected: optionSelected,
+            });
+          }
+
+          return (
+            <SelectOption
+              role="option"
+              a11yTitle={
+                value.includes(iValue)
+                  ? `${iLabel} selected`
+                  : `${iLabel} not selected`
+              }
+              aria-setsize={value.length}
+              aria-posinset={value.indexOf(iValue)}
+              aria-selected={value.includes(iValue)}
+              aria-disabled={isDisabled(indexOptions)}
+              plain
+              hoverIndicator
+              fill="horizontal"
+              tabIndex="0"
```

Comment:
It would be nice if when the checkbox receives keyboard focus via tabbing that the active checkbox receives the hover treatment so I can still see where focus is. (not sure if this is the right line to be placing this comment).

Video below, I am "shift+tabbing" back through the active options but can't see any indication of where I am until I reach the search:

https://user-images.githubusercontent.com/12522275/185683005-391464be-347b-4f45-b31d-ff7f04eefdfb.mov

Reply:
Good idea, I changed this in the latest commit